### ISSUE TITLE
SDK compile error with arm-none-eabi-gcc version 12.2.Rel1

### ISSUE THREAD

**Issue by whati001** [raspberrypi/pico-sdk#1416]
Hi, 

I am using Arch linux and use the package [gcc-arm-none-eabi-bin](https://aur.archlinux.org/packages/gcc-arm-none-eabi-bin) for compiling the SDK. Unfortunately, since upgrading my system, I am unable to compile the SDK code.

*gcc version*:
```
$ /usr/bin/arm-none-eabi-gcc --version
arm-none-eabi-gcc (Arm GNU Toolchain 12.2.Rel1 (Build arm-12.24)) 12.2.1 20221205
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

The problem is not trivial; adding the single include fixes the problem. But I want to record this finding within this issue ticket.
After pulling the currently latest PICO SDK version (f396d05f8252d4670d4ea05c8b7ac938ef0cd381) and running `cmake` and `make`, the following error occurs:
```
In file included from /home/user/pico-sdk/tools/pioasm/pio_disassembler.cpp:10:
/home/user/pico-sdk/tools/pioasm/pio_disassembler.h:16:25: error: ‘uint16_t’ was not declared in this scope
   16 | std::string disassemble(uint16_t inst, uint sideset_bits, bool sideset_opt);
      |                         ^~~~~~~~
/home/user/pico-sdk/tools/pioasm/pio_disassembler.h:1:1: note: ‘uint16_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
  +++ |+#include <cstdint>
    1 | /*
/home/user/pico-sdk/tools/pioasm/pio_disassembler.h:16:45: error: expected primary-expression before ‘sideset_bits’
   16 | std::string disassemble(uint16_t inst, uint sideset_bits, bool sideset_opt);
      |                                             ^~~~~~~~~~~~
/home/user/pico-sdk/tools/pioasm/pio_disassembler.h:16:59: error: expected primary-expression before ‘bool’
   16 | std::string disassemble(uint16_t inst, uint sideset_bits, bool sideset_opt);

```

As the error already indicates, we have to add the include `#include <cstdint>` to the file `pico-sdk/tools/pioasm/pio_disassembler.h`. After including the c-inttypes, the compilation works again.

```c
#ifndef _PIO_DISASSEMBLER_H
#define _PIO_DISASSEMBLER_H

#ifdef __cplusplus

#include <string>
#include <cstdint>

typedef unsigned int uint;

```

Please let me know if I should open a new pull request or if somebody includes this change.

**Comment by lurch** [comment]
Already fixed on the `develop` branch? #1367

**Comment by whati001** [comment]
Indeed, thank you very much.
